python/pyarrow/ipc.py

```diff
@@ -150,9 +140,8 @@ def _ensure_default_ipc_read_options(options):
     return options or IpcReadOptions()
 
 
-def new_stream(sink, schema, *, use_legacy_format=None, options=None):
+def new_stream(sink, schema, *, options=None):
```

Comment:
There hasn’t been a clean way to deprecate these so far. One approach could be to leave this PR open for now and introduce deprecation warnings in a separate PR first.

Reply:
This has been the name of the field for more than 5 years (`use_legacy_format` and the documentation said it was deprecated). Maybe we do not require to wait for a `DeprecationWarning` for a field that has been documented as deprecated for 5 years and has the name `legacy` on it? This is the commit that introduced the field on (Arrow 1.0 already had it as legacy)
https://github.com/apache/arrow/commit/1a7519f42096f3342224adf7f1643ab11d5098b3

Reply:
I'm a bit on the fence here, but I noticed that David has approved, and it looks like you, @raulcd, are on board as well. Unless there are any objections, I'll go ahead and merge this later today.